### PR TITLE
Added new hardware class for Denkovi devices

### PR DESCRIPTION
**Pull request by dzelyaydinov**
Support for DAEnet IP4, SmartDEN IP 16, SmartDEN IP 32 In, SmartDEN Logger, SmartDEN Watchdog, SmartDEN Notifier and SmartDEN IP-Maxi.

### CODE REVIEW

**Comment by gizmocuz** [comment]
Thanks for the patch! There are build errors, see the problem here above/below
Also when using parameters of a function that do not change, please declare them as 'std::string &parameter'

File: main/SQLHelper.cpp [hunk 1]
```diff
@@ -2427,6 +2427,7 @@ bool CSQLHelper::OpenDatabase()
 		{
 			safe_query("UPDATE Hardware SET Mode1 = 5000 WHERE Type = %d", HTYPE_DenkoviSmartdenIPInOut);
 			safe_query("UPDATE Hardware SET Mode1 = 5000 WHERE Type = %d", HTYPE_DenkoviSmartdenLan);
+			safe_query("UPDATE Hardware SET Mode1 = 5000 WHERE Type = %d", HTYPE_DenkoviDevices);
```
Comment (gizmocuz) [inline main/SQLHelper.cpp]: You need to increase the database version (to 130 as we speak) and make a new 'if' line for this change

Comment (dzelyaydinov) [reply main/SQLHelper.cpp]: Thanks for the remind. Is this mean to remove this line 
```
			safe_query("UPDATE Hardware SET Mode1 = 5000 WHERE Type = %d", HTYPE_DenkoviDevices);
```
and add
```
if (dbversion < 130)
		{
			safe_query("UPDATE Hardware SET Mode1 = 5000 WHERE Type = %d", HTYPE_DenkoviDevices);
		}
```
Thanks in advance.

Comment (gizmocuz) [reply main/SQLHelper.cpp]: yes it is, thanks !

File: main/SQLHelper.cpp [hunk 2]
```diff
@@ -2428,6 +2428,10 @@ bool CSQLHelper::OpenDatabase()
 			safe_query("UPDATE Hardware SET Mode1 = 5000 WHERE Type = %d", HTYPE_DenkoviSmartdenIPInOut);
 			safe_query("UPDATE Hardware SET Mode1 = 5000 WHERE Type = %d", HTYPE_DenkoviSmartdenLan);
 		}
+		if (dbversion < 130)
```
Comment (gizmocuz) [inline main/SQLHelper.cpp]: Something seems not right here
you placed a check for version 130 above 124 ?
Could you place it where it belongs ?
You also need to increase the actual database version with 1 at the top of the class

**Comment by gizmocuz** [comment]
Thanks for this great addon !
Shall i remove the following hardware types?:

HTYPE_DenkoviSmartdenLan:
HTYPE_DenkoviSmartdenIPInOut:

**Comment by gizmocuz** [comment]
Another question, your new hardware class is

HTYPE_DenkoviDevices

Then i think you do not have to patch the database setting the type to 5000, as it is a new hardware class correct ?

		if (dbversion < 130)
		{
			safe_query("UPDATE Hardware SET Mode1 = 5000 WHERE Type = %d", HTYPE_DenkoviDevices);
		}

**Comment by dzelyaydinov** [comment]
Thanks again for the support. Yes, you are right about this:
```
Then i think you do not have to patch the database setting the type to 5000, as it is a new hardware class correct ?
```
Please do not remove old hardware types if this will make trouble to our current clients. We just do not want our clients to lose their hardware and devices.

**Comment by gizmocuz** [comment]
@dzelyaydinov , true, but it is a one time job they have to do...
But maybe we will start removing the old classes after the new stable release

**Pull request merged**